pkg/nodediscovery/nodediscovery.go

```diff
@@ -183,13 +181,26 @@ func (n *NodeDiscovery) StartDiscovery(nodeName string) {
 		close(n.Registered)
 	}()
 
-	go func() {
+	if option.Config.JoinK8sNamespace != "" {
+		// Wait for successful node registration before
+		// informing the manager if registering to a k8s
+		// namespace as it can update node information
 		select {
 		case <-n.Registered:
 		case <-time.NewTimer(defaults.NodeInitTimeout).C:
 			log.Fatalf("Unable to initialize local node due to timeout")
 		}
-	}()
+	} else {
+		go func() {
+			select {
+			case <-n.Registered:
+			case <-time.NewTimer(defaults.NodeInitTimeout).C:
+				log.Fatalf("Unable to initialize local node due to timeout")
+			}
+		}()
+	}
+
+	n.Manager.NodeUpdated(n.LocalNode)
```

Comment:
The goroutine above that runs `RegisterNode(...)` would previously occur after this `NodeUpdated()` logic is run, but now there is no such guarantee. Is this OK?

Reply:
There is no guarantee of when a go routine is scheduled, so this should not change anything.

Reply:
Previously we had something like:

```
NodeUpdated()
go func()  { .... RegisterNode() ... }
```

Means there was a guarantee that `NodeUpdated()` returned before calling `RegisterNode()`, ie the local datapath is plumbed before propagating the node to the kvstore. That said I can't think of a specific case where letting remote nodes know about the local node before locally plumbing it would make a meaningful difference.